pkg/image/trigger/deploymentconfigs/deploymentconfigs.go

```diff
@@ -128,13 +128,19 @@ func (i deploymentConfigTriggerIndexer) Index(obj, old interface{}) (string, *tr
 	default:
 		// updated
 		dc = obj.(*appsapi.DeploymentConfig)
+		oldDC := old.(*appsapi.DeploymentConfig)
 		triggers = calculateDeploymentConfigTriggers(dc)
-		oldTriggers := calculateDeploymentConfigTriggers(old.(*appsapi.DeploymentConfig))
+		oldTriggers := calculateDeploymentConfigTriggers(oldDC)
 		switch {
 		case len(oldTriggers) == 0:
 			change = cache.Added
 		case !reflect.DeepEqual(oldTriggers, triggers):
 			change = cache.Updated
+		// We need to react on image changes as well. Image names could change,
+		// images could be set to different value or resetted to "" e.g. by oc apply
+		// and we need to make sure those changes get reconciled by re-resolving images
+		case !reflect.DeepEqual(dc.Spec.Template.Spec.Containers, oldDC.Spec.Template.Spec.Containers):
```

Comment:
should be check only the images then?

Reply:
worst case this will be processed but not triggered, but I think we need some common helper like HasUpdatedImages() to work well for this and for other places. this is fine now.

Reply:
discussed on IRC but for the record: I don't see other way we can determine if it needs to be queued without actually doing all the work in UpdateDeploymentConfigImages. This is just a check for adding to work queue, it should be fast and simple. After picking it up from queue we already do all the checks for updated images and others.

Reply:
What happens when I want to run my own image for sometime and not the image resolved by the trigger? Can I do that now?

Reply:
@kargakis set `automatic` to false
if it's active the state should always be reconciled